lms/djangoapps/grades/rest_api/v1/gradebook_views.py

```diff
@@ -515,22 +525,27 @@ def get(self, request, course_key):
             serializer = StudentGradebookEntrySerializer(entry)
             return Response(serializer.data)
         else:
-            filter_kwargs = {}
-            related_models = []
-            if request.GET.get('username_contains'):
-                filter_kwargs['user__username__icontains'] = request.GET.get('username_contains')
-                related_models.append('user')
+            q_objects = []
+            if request.GET.get('user_contains'):
+                search_term = request.GET.get('user_contains')
+
+                q_objects.append(
+                    Q(user__username__icontains=search_term) |
+                    Q(programcourseenrollment__program_enrollment__external_user_key__icontains=search_term) |
+                    Q(user__email__icontains=search_term)
+                )
             if request.GET.get('cohort_id'):
                 cohort = cohorts.get_cohort_by_id(course_key, request.GET.get('cohort_id'))
                 if cohort:
-                    filter_kwargs['user__in'] = cohort.users.all()
+                    q_objects.append(Q(user__in=cohort.users.all()))
```

Comment:
More efficient way to do this, though it's a little uglier: `user__cohortmembership__course_id=course_id,
            user__cohortmembership__course_user_group__name=cohort_id`